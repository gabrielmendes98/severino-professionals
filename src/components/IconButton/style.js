import styled, { css } from 'styled-components';
import MuiIconButton from '@material-ui/core/IconButton';
import { marginHandler } from 'commons/styles/margin';
import {
  getContrastColor,
  getColor,
  getDarkerColor,
} from 'commons/styles/util';
import { notForward } from 'commons/styles/config';

export const StyledIconButton = styled(MuiIconButton).withConfig(
  notForward('customColor'),
)`
  ${marginHandler};

  ${({ customColor }) =>
    customColor &&
    css`
      background-color: ${({ theme, customColor }) =>
        getColor(theme, customColor)};

      color: ${({ theme, customColor }) =>
        getContrastColor(theme, customColor)};

      &:hover {
        background-color: ${({ theme, customColor }) =>
          getDarkerColor(theme, customColor)};
    `}
`;
