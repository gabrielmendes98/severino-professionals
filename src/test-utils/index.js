/* eslint-disable react/prop-types */
import { render, configure, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Form, Formik } from 'formik';
import MainProvider from 'commons/providers/MainProvider';

configure({ testIdAttribute: 'id' });

const Wrapper = ({ children }) => (
  <>
    <MainProvider>{children}</MainProvider>
    <div id="modal"></div>
  </>
);
const RouteWrapper = ({ children }) => (
  <BrowserRouter>
    <Wrapper>{children}</Wrapper>
  </BrowserRouter>
);
const FormikWrapper = ({ children, initialValues, ...config }) => (
  <Formik initialValues={initialValues} {...config}>
    <Form>{children}</Form>
  </Formik>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

const renderWithRouter = (ui, { route = '/', ...options } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: RouteWrapper, ...options });
};

const asFragment = ui => {
  const { asFragment } = customRender(ui);
  return asFragment(ui);
};

const toMatchSnapshot = (Component, { useRouter = false } = {}) => {
  const wrapper = asFragment(
    useRouter ? <BrowserRouter>{Component}</BrowserRouter> : Component,
  );
  expect(wrapper).toMatchSnapshot();
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export {
  customRender as render,
  screen,
  toMatchSnapshot,
  asFragment,
  renderWithRouter,
  FormikWrapper,
};
