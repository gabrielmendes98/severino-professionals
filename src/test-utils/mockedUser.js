const mockedUser = {
  id: '9c6cb22e-f284-4ca4-9a88-d64ece7b896c',
  createdAt: '2021-09-02T03:55:52.338Z',
  updatedAt: '2021-09-02T03:55:52.338Z',
  email: 'test@test.com',
  name: 'Testerson da Silva',
  description: null,
  avatarUrl: 'www.avatarurlfake.com',
  phone: '(11) 11111-1112',
  state: 'Minas Gerais',
  city: '√Åguas Formosas',
  hasWhatsapp: false,
};
export default mockedUser;
