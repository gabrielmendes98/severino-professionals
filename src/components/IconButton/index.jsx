import PropTypes from 'prop-types';
import Tooltip from '@material-ui/core/Tooltip';
import { StyledIconButton } from './style';
import { isDefaultColor } from './util';

const IconButton = ({ children, color, tooltip, ...props }) => {
  const { defaultColor, customColor } = isDefaultColor(color)
    ? { defaultColor: color }
    : { customColor: color };

  return (
    <Tooltip title={tooltip}>
      <StyledIconButton
        {...props}
        color={defaultColor}
        customColor={customColor}
        aria-label={tooltip}
      >
        {children}
      </StyledIconButton>
    </Tooltip>
  );
};

IconButton.propTypes = {
  tooltip: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  color: PropTypes.string,
  margin: PropTypes.shape({
    top: PropTypes.number,
    bottom: PropTypes.number,
    right: PropTypes.number,
    left: PropTypes.number,
  }),
};

IconButton.defaultProps = {
  variant: 'contained',
  color: 'primary',
};

export default IconButton;
