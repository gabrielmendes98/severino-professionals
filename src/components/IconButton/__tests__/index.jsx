import { toMatchSnapshot } from 'test-utils';
import AcUnitIcon from '@material-ui/icons/AcUnit';
import IconButton from '..';

it('should match snapshot', () => {
  toMatchSnapshot(
    <IconButton tooltip="test tooltip">
      <AcUnitIcon />
    </IconButton>,
  );
});

it('should match snapshot with custom color', () => {
  toMatchSnapshot(
    <IconButton tooltip="test tooltip " color="red">
      <AcUnitIcon />
    </IconButton>,
  );
});
