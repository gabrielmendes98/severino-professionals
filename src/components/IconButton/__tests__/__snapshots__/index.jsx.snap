// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
}

<button
    aria-label="test tooltip"
    class="MuiButtonBase-root MuiIconButton-root c0 MuiIconButton-colorPrimary"
    tabindex="0"
    title="test tooltip"
    type="button"
    variant="contained"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    id="modal"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`should match snapshot with custom color 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  background-color: #F44336;
  color: #fff;
}

.c0:hover {
  background-color: rgb(195,53,43);
}

<button
    aria-label="test tooltip "
    class="MuiButtonBase-root MuiIconButton-root c0"
    tabindex="0"
    title="test tooltip "
    type="button"
    variant="contained"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    id="modal"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
